### PR TITLE
Issue 372: expand tests and coverage for DateTimeIso8601 serializer t…

### PR DESCRIPTION
**Pull request by jmctwilio**
…o support strings and null.

### CODE REVIEW

File: src/Twilio/Converters/Serializers.cs [hunk 1]
```diff
@@ -17,5 +19,27 @@ public static string JsonObject(object input)
...
+        /// <param name="input">DateTime intance to serialize to string</param>
+        /// <returns>A string</returns>
+        public static string DateTimeIso8601(DateTime input)
+        {
+              return input.ToString("yyyy-MM-ddTHH:mm:ssZ");
+        }
+
+        public static string DateTimeIso8601(string input)
+        {
+            CultureInfo enUS = new CultureInfo("en-US");
```
Comment (ilanbiala) [inline src/Twilio/Converters/Serializers.cs]: Do we always want to use en-US?

File: src/Twilio/Converters/Serializers.cs [hunk 2]
```diff
@@ -17,5 +19,29 @@ public static string JsonObject(object input)
...
+        /// <summary>
+        /// Produce a ISO 8601 compatible string from input if possible
+        /// </summary>
+        /// <param name="input">DateTime intance to serialize to string</param>
+        /// <returns>A string</returns>
+        public static string DateTimeIso8601(DateTime input)
+        {
+              return input.ToString("yyyy-MM-ddTHH:mm:ssZ");
+        }
+
```
Comment (jingming) [inline src/Twilio/Converters/Serializers.cs]: Add XML docs.

File: src/Twilio/Converters/Serializers.cs [hunk 3]
```diff
@@ -17,5 +19,29 @@ public static string JsonObject(object input)
...
+        public static string DateTimeIso8601(string input)
+        {
+            if (input == null) return null;
+
+            CultureInfo enUS = new CultureInfo("en-US");
```
Comment (jingming) [inline src/Twilio/Converters/Serializers.cs]: You can do `var variableName` for all the variables.

File: src/Twilio/Converters/Serializers.cs [hunk 4]
```diff
@@ -17,5 +19,29 @@ public static string JsonObject(object input)
...
+            DateTimeStyles utc = DateTimeStyles.AdjustToUniversal;
+            DateTime parsedDateTime;
+
+            if (DateTime.TryParse(input, enUS, utc, out parsedDateTime))
```
Comment (jingming) [inline src/Twilio/Converters/Serializers.cs]: Does this work as a ternary? 
```
return DateTime.TryParse(input, enUS, utc, out parsedDateTime) ? DateTimeIso8601(parsedDateTime) : input;
```

File: src/Twilio/Converters/Serializers.cs (hunk 1 above)
Comment (efossier) [inline src/Twilio/Converters/Serializers.cs]: Yes because all our apis expect US culture dates etc

File: src/Twilio/Converters/Serializers.cs [hunk 5]
```diff
@@ -17,5 +19,31 @@ public static string JsonObject(object input)
...
+        /// <summary>
+        /// Produce a ISO 8601 UTC compatible string from input if possible
+        /// </summary>
+        /// <param name="input">string representation of a time which will be converted to an iso8601 UTC string</param>
+        /// <returns>A string</returns>
+        public static string DateTimeIso8601(string input)
+        {
+            if (input == null) return null;
+
+            var enUS = new CultureInfo("en-US");
```
Comment (jingming) [inline src/Twilio/Converters/Serializers.cs]: Can you pass these two in directly into the function call?

Comment (efossier) [inline src/Twilio/Converters/Serializers.cs]: sure

Comment (efossier) [inline src/Twilio/Converters/Serializers.cs]: Actually that makes the line really long so im gonna leave it as is.

**Pull request merged**